pkg/internal/computeresources/tasklevel/tasklevel_test.go

```diff
@@ -0,0 +1,236 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tasklevel_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/internal/computeresources/tasklevel"
+	"github.com/tektoncd/pipeline/test/diff"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestPodBuild_TaskLevelResourceRequirements(t *testing.T) {
+	testcases := []struct {
+		desc                     string
+		ts                       v1beta1.TaskSpec
```

Comment:
Since the function being tested accepts a slice of steps, I think that's what should be in the test case. (I don't think it makes sense to use a TaskSpec that has other fields including sidecars, because this function doesn't accept task specs or sidecars in its input.

Reply:
the tested function here we mean `builder.Build()` right? which needs a TaksRun and a TaskSepc

Reply:
the `Build` function is the one we mentioned in last week to test as it covers logic related to steps, stepTemplate, sidecar to form the container configs to create the Pod. 
`verifyTaskLevelComputeResources` is the help func which we need to verify if the result pod matches our expectation.

Reply:
It looks like this function is testing ApplyTaskLevelResourceRequirements not builder.Build

Reply:
I think your approach in pod_test.go (where you're using the TaskSpec and calling builder.Build) is correct!

Reply:
Unresolving this conversation, because while the tests in pod_test.go LGTM (the ones testing builder.Build), this test (in tasklevel_test.go) is testing the function `tasklevel.ApplyTaskLevelComputeResources`, which only takes steps and compute resources. Therefore, I think the test cases in this file should use steps and compute resources as their inputs, not task specs with other fields in addition.

Reply:
missed these 2 files. sry for this. 
thanks the double check - let me update!

Reply:
cleaned testcases struct.